composer/datasets/in_context_learning_evaluation.py

```diff
@@ -1159,78 +890,97 @@ def build_icl_dataloader(
     prompt_string: str,  # e.g. 'translate english to french:'
     example_delimiter: str,  # e.g. '\n'
     continuation_delimiter: str,  # e.g. ''
+    hf_loading_vars: dict,
+    hf_parsing_vars: dict,
     destination_path: str,
-    question_prelimiter: str = '',  # e.g. 'Question: '
-    cot_delimiter: str = '',
-    fewshot_random_seed: int = 1234,
-    pass_at_k: int = 1,
-    generations_per_sample: int = 1,
+    prelimiter: str,  # e.g. 'Question: '
+    cot_delimiter: str,
+    fewshot_random_seed: int,
+    pass_at_k: int,
+    generations_per_sample: int,
+    hf_parsing_func: Callable = None,
```

Comment:
any task specific args here should default to `None` and then raise an error in the specific if case they are needed for if they are still `None`

Reply:
To be clear, we want to force any task specific arg that can affect the output of the eval (i.e. changes the metrics of the task like pass_at_k or generations_per_sample rather than change the parsing of the file like context_key or stacked_keys) to be user specified rather than have defaults?